packages/frontend-2/components/server-management/UserRoleSelect.vue

```diff
@@ -0,0 +1,53 @@
+<template>
+  <FormSelectBase
+    v-model="selectedValue"
+    :items="roles"
+    :label="'Role'"
+    :name="'userRole'"
+    :allow-unset="false"
+  >
+    <template #something-selected="{ value }">
+      <span class="truncate">{{ getRoleLabel(value) }}</span>
+    </template>
+
+    <template #option="{ item }">
+      <span class="truncate">{{ getRoleLabel(item) }}</span>
+    </template>
+  </FormSelectBase>
+</template>
+
+<script setup lang="ts">
+import { Roles, ServerRoles } from '@speckle/shared/src/core/constants'
+import { useFormSelectChildInternals } from '~~/lib/form/composables/select'
+
+type ValueType = ServerRoles | ServerRoles[] | undefined
+
+type Role = ServerRoles
+
+const props = defineProps<{
+  modelValue?: ValueType
+  multiple?: boolean
+}>()
+
+const emit = defineEmits<{
+  (e: 'update:modelValue', v: ValueType): void
+}>()
+
+const { selectedValue } = useFormSelectChildInternals<ServerRoles>({
+  props: toRefs(props),
+  emit
+})
+
+const roleLookupTable = {
```

Comment:
If you put this in a helper and export it out, you can use it elsewhere as well (e.g. that Role dialog where you needed these names)